src/TwilioVideo.android.js

```diff
@@ -309,6 +325,35 @@ class CustomTwilioVideoView extends Component {
       "onLocalParticipantSupportedCodecs",
     ].reduce((wrappedEvents, eventName) => {
       if (this.props[eventName]) {
+        // if (eventName === "onDataTrackBinaryMessageReceived") {
+        if (eventName === "onDataTrackMessageReceived") {
+          return {
+            ...wrappedEvents,
+            [eventName]: (data) => {
+              console.log(
+                // "HEYYY onDataTrackBinaryMessageReceived, deserializing"
+                "HEYYY SDK onDataTrackMessageReceived, deserializing"
+              );
+              console.log(
+                `data.nativeEvent.message: ${data.nativeEvent.message}`
+              );
+              // console.log(typeof data.nativeEvent.message); // string
+              var uint8 = Uint8Array.from(
+                data.nativeEvent.message.split("").map((x) => x.charCodeAt())
+              );
+              console.log(`UInt8 value: ${uint8}`);
+              // console.log(typeof uint8);  // object
+              //Deserialize here
+              //Do a base64 decode on data.nativeEvent.message to turn into UTF-8
+              // const decodedString = decode(data.nativeEvent.message);
+              // console.log(decodedString);
+              // this.props[eventName](decodedString);
+              this.props[eventName](data.nativeEvent);
```

Comment:
If you use the `base64-js` library I mentioned above, then all you should need is:

```js
this.props[eventName]({
    ...data.nativeEvent,
    message: base64js.toByteArray(data.nativeEvent.message),
});
```